### DISCUSSION TITLE
MoveTo node

### DISCUSSION THREAD

**Discussion by skadefro** [bcakmakoglu/vue-flow#1242] [Q&A] [answered]
I feel stupid, but i cannot for the life of me figure out how to move the viewport to a specific node ?
If i have an instance declared with
```
const instance = useVueFlow({})
```
and i get a node with
```
const node = instance.findNode(id)
```
How do i move the viewport so the node is in the center of the view ?

Example: I zoom in on a node, and i can see 
instance.viewport.value.x is 536 
instance.viewport.value.y is198
but the node's x is -240 and y is 60
so if i use 
```
instance.setViewport({ x: node.position.x, y: node.position.y, zoom: 1 });
```
I land somewhere completely different.

I tried to see if adding/removing instance.dimensions / instance.vueFlowRef.value.getBoundingClientRect and others in, but i simply cannot "get" the logic, on how to map back and forth from x,y  on a node to what setViewport want's

**Comment by bcakmakoglu** [comment, marked as answer]
The simplest way to do this is to use `fitView`

```ts
const { fitView } = useVueFlow();

function zoomToNode(nodeId: string) {
	fitView({
		nodes: [nodeId],
		duration: 1000, // use this if you want a smooth transition to the node
		padding: 1 // use this for some padding around the node
	})
}

**Comment by skadefro** [comment]
that looks awesome ... 
Thank you :-)

**Comment by 0xAnon0602** [comment]
@skadefro 
This doesn't work for me 
It tries to focus only on specific nodes and not on all the nodes I've provided

**Reply by bcakmakoglu** [reply]
Can you provide some code / a repro for the issue?
Are you sure you're not limiting fit-view by some max. zoom that is present (either the default or a value you have set yourself)?

**Reply by dbomfim** [reply]
@0xAnon0602 If you provide just the id of the node you want, it will work:

```javascript
const { onPaneReady, fitView } = useVueFlow()

const initialNodes = [
  {
    id: '1',
    type: 'input',
    data: { label: 'Node 1' },
    position: { x: 250, y: 0 },
    class: 'light',
  },
  {
    id: '2',
    type: 'output',
    data: { label: 'Node 2' },
    position: { x: 100, y: 100 },
    class: 'light',
  },
]

onPaneReady((pane) => {
  fitView({
    nodes: ['1'],
    duration: 1000, // use this if you want a smooth transition to the node
    padding: 1,
  })
})


```
